src/algebra/flatten_tuple.jl

```diff
@@ -0,0 +1,6 @@
+# type stable flatten function for tuples
+flatten_tuple() = ()
+flatten_tuple(a) = (a,)
+flatten_tuple(a::Tuple, b...) = tuple(flatten_tuple(a...)..., flatten_tuple(b...)...)
+flatten_tuple(a, b...) = tuple(a, flatten_tuple(b...)...)
+flatten_tuple(x::Tuple) = tuple(flatten_tuple(x...)...)
```

Comment:
This can be implemented more simply as
```suggestion
flatten(x, y...) = (flatten(x)..., flatten(y...)...)
flatten(x::Tuple) = flatten(x...)
flatten(x) = (x,)
```

Reply:
Imported the flatten function from `TupleTools.jl`. Resolved by 29f246b.